humility-core/src/core.rs

```diff
@@ -726,38 +794,69 @@ impl Core for OpenOCDCore {
     }
 
     fn halt(&mut self) -> Result<()> {
+        log::trace!("halting core");
         //
         // On OpenOCD, we don't halt. If GDB is connected, it gets really,
         // really confused!  This should probably be configurable at
         // some point...
         //
+        // Well without unhalted read support cant do anything,
```

Comment:
It theoretically does, but when humility and GDB are both connected to openocd, there is lots of other weird behavior.  With the current setup, everything should work as expected as long as you halt the core in GDB before running humility.  This behavior may be different on arm though, as it supports unhalted access.

Reply:
Testing changes to the `cores` is difficult, so everything is not rigorously tested on arm.  I am thinking of setting up a test suite that requires you to connect different boards, as the current test suite does not exercise these cores.

Reply:
Makes sense. Yeah, as we start pushing this upstream, ensuring we do not regress any of the other boards would be critical. Of course if the arm boards don't work currently with ToT, it would be difficult to check for regressions.